File: programmers/3진법 뒤집기.py
def solution(n):
    answer = 0
    temp=''
    m=1
    if n<3:
        return n
    else:
        while True:
            temp=str(n%3)+temp
            n=n//3   
            if n<3:
                temp=str(n%3)+temp
                break
        for i in temp:
            answer+=int(i)*m
            m=m*3

        return answer

File: programmers/3진법 뒤집기.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public int solution(int n) {
        int answer = 0;
        ArrayList<Integer> box = new ArrayList<>();
        int r = 0;
        int q = n;
        
        if(n<3){
            answer = n;
        }
        else{
            while (true){
                r = q%3;
                q = q/3;
                if(q<3){
                    box.add(r);
                    box.add(q);
                    break;
                }
                else{
                    box.add(r);
                }
            }
        }
        for(int i=0;i<box.size();i++){
            
            answer += box.get(i)*Math.pow(3, box.size()-i-1);

        }
        return answer;
    }
}